runtime/Java/src/org/antlr/v4/runtime/tree/xpath/XPathLexer.java

```diff
@@ -0,0 +1,179 @@
+		for (int i = 0; i < tokenNames.length; i++) {
+			tokenNames[i] = VOCABULARY.getLiteralName(i);
+			if (tokenNames[i] == null) {
+				tokenNames[i] = VOCABULARY.getSymbolicName(i);
+			}
+
+			if (tokenNames[i] == null) {
+				tokenNames[i] = "<INVALID>";
+			}
+		}
+	}
+
+	@Override
+	public String getGrammarFileName() { return "XPathLexer.g4"; }
+
+	@Override
+	public String[] getRuleNames() { return ruleNames; }
+
+	@Override
+	public String[] getModeNames() { return modeNames; }
+
+	@Override
+	@Deprecated
+	public String[] getTokenNames() {
+		return tokenNames;
+	}
+
+	@Override
+	public Vocabulary getVocabulary() {
+		return VOCABULARY;
+	}
+
+	@Override
+	public ATN getATN() {
+		return null;
+	}
+
+	protected int line = 1;
+	protected int charPositionInLine = 0;
+
+	public XPathLexer(CharStream input) {
+		super(input);
+	}
+
+	@Override
+	public Token nextToken() {
+		_tokenStartCharIndex = _input.index();
+		CommonToken t = null;
+		while ( t==null ) {
+			switch ( _input.LA(1) ) {
+				case '/':
+					consume();
+					if ( _input.LA(1)=='/' ) {
+						consume();
+						t = new CommonToken(ANYWHERE, "//");
+					}
+					else {
+						t = new CommonToken(ROOT, "/");
+					}
+					break;
+				case '*':
+					consume();
+					t = new CommonToken(WILDCARD, "*");
+					break;
+				case '!':
+					consume();
+					t = new CommonToken(BANG, "!");
+					break;
+				case '\'':
+					String s = matchString();
+					t = new CommonToken(STRING, s);
+					break;
+				case CharStream.EOF :
+					return new CommonToken(EOF, "<EOF>");
+				default:
+					if ( isNameStartChar(_input.LA(1)) ) {
+						String id = matchID();
+						if ( Character.isUpperCase(id.charAt(0)) ) t = new CommonToken(TOKEN_REF, id);
+						else t = new CommonToken(RULE_REF, id);
+					}
+					else {
+						throw new LexerNoViableAltException(this, _input, _tokenStartCharIndex, null);
+					}
+					break;
+			}
+		}
+		t.setStartIndex(_tokenStartCharIndex);
+		t.setCharPositionInLine(_tokenStartCharIndex);
+		t.setLine(line);
+		return t;
+	}
+
+	public void consume() {
+		int curChar = _input.LA(1);
+		if ( curChar=='\n' ) {
+			line++;
+			charPositionInLine=0;
+		}
+		else {
+			charPositionInLine++;
+		}
+		_input.consume();
+	}
+
+	@Override
+	public int getCharPositionInLine() {
+		return charPositionInLine;
+	}
+
+	public String matchID() {
+		int start = _input.index();
+		consume(); // drop start char
+		while ( isNameChar(_input.LA(1)) ) {
+			consume();
+		}
+		return _input.getText(Interval.of(start,_input.index()-1));
+	}
+
+	public String matchString() {
+		int start = _input.index();
+		consume(); // drop first quote
+		while ( _input.LA(1)!='\'' ) {
+			consume();
+		}
+		consume(); // drop last quote
+		return _input.getText(Interval.of(start,_input.index()-1));
+	}
+
+	public boolean isNameChar(int c) { return Character.isUnicodeIdentifierPart(c); }
+
+	public boolean isNameStartChar(int c) { return Character.isUnicodeIdentifierStart(c); }
```

Comment:
GWT does not have `isUnicodeIdentifierPart` or `isUnicodeIdentifierStart` for `Character`.

Reply:
@aranega By the way if you have any ideas how to get this to work in GWT I welcome them. Searching for a solution, but don't one yet.